replaced a tag with Link and updated tests

tanwarAalok opened the pull request:
## Description
#549 Replaced the <a> tag with <Link> tag in Header and Footer and updated the Header and Footer test accordingly.

## What type of PR is this? (Check all applicable)

- [ ] 🍕 Feature
- [x] 🐛 Bug Fix
- [ ] 📄 Documentation Update
- [ ] 👨‍💻 Code Refactor
- [x] 🔥 Performance Improvements
- [ ] ✅ Test
- [ ] 🛠️ CI/CD

## Screenshots (if applicable)


https://github.com/user-attachments/assets/5598117c-5672-4827-969e-d93faf00629a



![image](https://github.com/user-attachments/assets/3d1c40a7-85a4-4263-8204-9ef9228cc38b)
![image](https://github.com/user-attachments/assets/6178f0dd-ac55-42e0-a4c5-d56bad6b5998)



## Checklist
- [x] I have performed a self-review of my code  
- [ ] I have commented my code, particularly in hard-to-understand areas  
- [x] I have added tests that prove my fix is effective or that my feature works  
- [x] New and existing unit tests pass locally with my changes

tanwarAalok committed:
replaced a tag with Link and updated tests

tanwarAalok committed:
docs route uncommented

aps08 committed:
Merge branch 'dev' into bug/549-replace-anchor-tag

aps08 committed:
Merge branch 'dev' into bug/549-replace-anchor-tag

aps08 committed:
Merge branch 'dev' into bug/549-replace-anchor-tag

aps08 committed:
Merge branch 'dev' into bug/549-replace-anchor-tag

tanwarAalok committed:
Merge branch 'dev' into bug/549-replace-anchor-tag

tanwarAalok committed:
Merge branch 'stage' into bug/549-replace-anchor-tag

aps08 reviewed (changes requested):
Nav is redirecting to different page even when items are on the home page. It should scroll to the section.

tanwarAalok committed:
section navigation fixed on header and footer

tanwarAalok commented:
@aps08  now navigation is working fine. I have added id to all the section where navigation was there and scrolling there. I have also added ScrolltoTop component to scroll to top whenever route changed because if I was on certain section in home page apart from top and routed to new route like /doc the /doc was not automatically at relative scrolled positioned as it was on previous route.

I didn't understand how should I proceed with testing section navigation as these are not directly part of Header or Footer component. So I left it for now with TODO comment which is why I am getting 2 issue in quality check.

tanwarAalok committed:
header and footer test updated

aps08 commented on packages/ui/src/components/footer/Footer.jsx:
```diff
@@ -1,24 +1,39 @@
+
   return (
-    <div className={`container ${styles.block}`}>
+    <div role="footer" className={`container ${styles.block}`}>
```
what is role here ?

tanwarAalok replied:
I am using role so that I can point out to footer in test 

``` const footerElement = screen.getByRole("footer");```

aps08 commented on packages/ui/src/components/header/Header.jsx:
```diff
@@ -23,7 +24,7 @@ const Header = () => {
 
   return (
-    <div className={`container ${styles.block}`}>
+    <div role="header" className={`container ${styles.block}`}>
```
what is role here ?

mayank2424 commented on packages/ui/src/utils/Constants.js:
```diff
@@ -94,7 +94,7 @@ export const HEADER_ITEMS = [
     name: "home",
     title: "Home",
-    url: "/",
+    url: "#home",
```
What's the reason to change it to `#`?

tanwarAalok replied:
If a user navigates to other sections like Features or Pricing and then clicks the Home button, it should scroll back to the top of the page. Using "/home" won't work in this case because the user is already on the homepage.

aps08 committed:
Merge branch 'stage' into bug/549-replace-anchor-tag

aps08 commented:
pre commit failing

tanwarAalok committed:
name updated acccording to aria role

tanwarAalok committed:
using data-testid in place of role

tanwarAalok commented:
> pre commit failing

fixed

aps08 committed:
Merge branch 'stage' into bug/549-replace-anchor-tag

aps08 committed:
prettier fix

tanwarAalok committed:
mobileheadertest updated

aps08 committed:
Merge branch 'stage' into bug/549-replace-anchor-tag

aps08 committed:
Merge branch 'stage' into bug/549-replace-anchor-tag

aps08 committed:
Merge branch 'stage' into bug/549-replace-anchor-tag

aps08 committed:
Merge branch 'stage' into bug/549-replace-anchor-tag

aps08 committed:
Merge branch 'stage' into bug/549-replace-anchor-tag
